crates/common/tedge_config/src/tedge_config_cli/settings.rs

```diff
@@ -457,6 +457,19 @@ impl ConfigSetting for RunPathSetting {
     type Value = FilePath;
 }
 
+pub struct DataPathSetting;
+
+impl ConfigSetting for DataPathSetting {
+    const KEY: &'static str = "data.path";
+
+    const DESCRIPTION: &'static str = concat!(
+        "The directory path to be used to store all persistent data like cached files, runtime metadata etc.",
```

Comment:
I don't think we need to mention "persistent" here, as the user can configure the data path to be stored a non-persistent path.

```suggestion
        "The directory path to be used to store all data like cached files, runtime metadata etc.",
```

Reply:
I wanted some word to distinguish it from "temporary" data. Any suggestions?

Reply:
I don't think we need to call it anything. We have examples of the data we are storing, so we can leave it as that. Let the user control whether it is persisted or not by their choice of location. (though obviously if they choose a volatile location, then they are responsible for initialising the structure, but this can be done using standard service dependencies)